packages/dev/core/src/Layers/highlightLayer.ts

```diff
@@ -625,15 +629,27 @@ export class HighlightLayer extends EffectLayer {
 
         const meshExcluded = this._excludedMeshes[mesh.uniqueId];
         if (!meshExcluded) {
-            this._excludedMeshes[mesh.uniqueId] = {
+            const obj: IHighlightLayerExcludedMesh = {
                 mesh: mesh,
-                beforeBind: mesh.onBeforeBindObservable.add((mesh: Mesh) => {
-                    mesh.getEngine().setStencilBuffer(false);
-                }),
-                afterRender: mesh.onAfterRenderObservable.add((mesh: Mesh) => {
-                    mesh.getEngine().setStencilBuffer(true);
-                }),
+                beforeBind: null,
+                afterRender: null,
+                stencilState: false,
             };
+
+            obj.beforeBind = mesh.onBeforeBindObservable.add(
+                function (this: IHighlightLayerExcludedMesh, mesh: Mesh) {

```

Comment:
I think you can prevent the cost of the bind by keeping the arrow function and using obj. instead of this ?